File: 2017/02/02.py
with open('input', 'r') as f:
    content = f.read()

    # One liner
    print(sum([max(r) - min(r) for r in [[int(i) for i in l.split('\t')] for l in content.splitlines()]]))

    # Part one
    s = 0
    for line in content.splitlines():
        ints = [int(i) for i in line.split('\t')]
        s += max(ints) - min(ints)
    print(s)

    # Part two
    s = 0
    for line in content.splitlines():
        ints = [int(i) for i in line.split('\t')]
        s += [(x / y) for x in ints for y in ints if x != y and x > y and x % y == 0][0]
    print(s)

File: 2017/02/02.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
	"strconv"
	"strings"
)

func main() {
	ints := ReadTabSeparatedLinesInts("input")

	fmt.Println(partOne(ints))
	fmt.Println(partTwo(ints))
}

func ReadTabSeparatedLinesInts(fileName string) [][]int {
	f, err := os.Open(fileName)
	if err != nil {
		panic(err)
	}
	defer f.Close()

	s := bufio.NewScanner(f)
	s.Split(bufio.ScanLines)

	var ints [][]int
	for s.Scan() {
		line := lineToInts(s.Text(), "\t")
		ints = append(ints, line)
	}

	return ints
}

func lineToInts(line, separator string) []int {
	var ints []int

	parts := strings.Split(line, separator)
	for _, a := range parts {
		i, err := strconv.Atoi(a)
		if err != nil {
			panic(err)
		}
		ints = append(ints, i)
	}
	return ints
}

func partOne(ints [][]int) int {
	sum := 0
	for _, line := range ints {
		min, max := math.MaxUint32, 0
		for _, i := range line {
			if i < min {
				min = i
			}
			if i > max {
				max = i
			}
		}
		diff := max - min
		sum += diff
	}
	return sum
}

func partTwo(ints [][]int) int {
	sum := 0
	for _, line := range ints {
	numberLabel:
		for i := 0; i < len(line); i++ {
			for j := i + 1; j < len(line); j++ {
				a, b := line[i], line[j]
				if a%b == 0 {
					sum += a / b
					break numberLabel
				} else if b%a == 0 {
					sum += b / a
					break numberLabel
				}
			}
		}
	}
	return sum
}
